2025-05-13 04:38:43,203 | INFO | üöÄ Starting ETL pipeline...
2025-05-13 04:38:43,204 | INFO | üöÄ Starting multiprocessing with 8 workers...
2025-05-13 04:38:43,967 | INFO | üìÑ Starting processing file: 2025 - 01 - 03- Night Audit Packet.pdf
2025-05-13 04:38:43,967 | INFO | üìÑ Starting processing file: 2025 - 01 - 02- Night Audit Packet.pdf
2025-05-13 04:38:43,967 | INFO | üìÑ Starting processing file: 2025 - 01 - 01- Night Audit Packet.pdf
2025-05-13 04:38:43,967 | INFO | üìÑ Starting processing file: 2025 - 01 - 04- Night Audit Packet.pdf
2025-05-13 04:38:43,967 | INFO | üìÑ Starting processing file: 2025 - 01 - 05- Night Audit Packet.pdf
2025-05-13 04:38:43,974 | INFO | üìÑ Starting processing file: 2025 - 01 - 06- Night Audit Packet.pdf
2025-05-13 04:38:43,987 | INFO | üìÑ Starting processing file: 2025 - 01 - 07- Night Audit Packet.pdf
2025-05-13 04:38:44,003 | INFO | üìÑ Starting processing file: 2025 - 01 - 08- Night Audit Packet.pdf
2025-05-13 04:39:00,770 | ERROR | ‚ùå A/R Aging extraction failed: 'Engine' object has no attribute 'cursor'
Traceback (most recent call last):
  File "/Users/sandeepmalkanoor/Documents/Python/Night_Audit_DataEngineering_Project/night_audit_etl_pipeline/processor.py", line 92, in handle_section
    insert_dataframe(engine, df, table_name, filename)
  File "/Users/sandeepmalkanoor/Documents/Python/Night_Audit_DataEngineering_Project/night_audit_etl_pipeline/db_utils.py", line 56, in insert_dataframe
    df.to_sql(table_name, con=engine, if_exists='append', index=False)
  File "/Users/sandeepmalkanoor/Documents/Python/airflow_env/lib/python3.11/site-packages/pandas/util/_decorators.py", line 333, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sandeepmalkanoor/Documents/Python/airflow_env/lib/python3.11/site-packages/pandas/core/generic.py", line 3087, in to_sql
    return sql.to_sql(
           ^^^^^^^^^^^
  File "/Users/sandeepmalkanoor/Documents/Python/airflow_env/lib/python3.11/site-packages/pandas/io/sql.py", line 842, in to_sql
    return pandas_sql.to_sql(
           ^^^^^^^^^^^^^^^^^^
  File "/Users/sandeepmalkanoor/Documents/Python/airflow_env/lib/python3.11/site-packages/pandas/io/sql.py", line 2850, in to_sql
    table.create()
  File "/Users/sandeepmalkanoor/Documents/Python/airflow_env/lib/python3.11/site-packages/pandas/io/sql.py", line 984, in create
    if self.exists():
       ^^^^^^^^^^^^^
  File "/Users/sandeepmalkanoor/Documents/Python/airflow_env/lib/python3.11/site-packages/pandas/io/sql.py", line 970, in exists
    return self.pd_sql.has_table(self.name, self.schema)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sandeepmalkanoor/Documents/Python/airflow_env/lib/python3.11/site-packages/pandas/io/sql.py", line 2865, in has_table
    return len(self.execute(query, [name]).fetchall()) > 0
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sandeepmalkanoor/Documents/Python/airflow_env/lib/python3.11/site-packages/pandas/io/sql.py", line 2672, in execute
    cur = self.con.cursor()
          ^^^^^^^^^^^^^^^
AttributeError: 'Engine' object has no attribute 'cursor'

2025-05-13 04:39:01,485 | ERROR | ‚ùå A/R Aging extraction failed: 'Engine' object has no attribute 'cursor'
Traceback (most recent call last):
  File "/Users/sandeepmalkanoor/Documents/Python/Night_Audit_DataEngineering_Project/night_audit_etl_pipeline/processor.py", line 92, in handle_section
    insert_dataframe(engine, df, table_name, filename)
  File "/Users/sandeepmalkanoor/Documents/Python/Night_Audit_DataEngineering_Project/night_audit_etl_pipeline/db_utils.py", line 56, in insert_dataframe
    df.to_sql(table_name, con=engine, if_exists='append', index=False)
  File "/Users/sandeepmalkanoor/Documents/Python/airflow_env/lib/python3.11/site-packages/pandas/util/_decorators.py", line 333, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sandeepmalkanoor/Documents/Python/airflow_env/lib/python3.11/site-packages/pandas/core/generic.py", line 3087, in to_sql
    return sql.to_sql(
           ^^^^^^^^^^^
  File "/Users/sandeepmalkanoor/Documents/Python/airflow_env/lib/python3.11/site-packages/pandas/io/sql.py", line 842, in to_sql
    return pandas_sql.to_sql(
           ^^^^^^^^^^^^^^^^^^
  File "/Users/sandeepmalkanoor/Documents/Python/airflow_env/lib/python3.11/site-packages/pandas/io/sql.py", line 2850, in to_sql
    table.create()
  File "/Users/sandeepmalkanoor/Documents/Python/airflow_env/lib/python3.11/site-packages/pandas/io/sql.py", line 984, in create
    if self.exists():
       ^^^^^^^^^^^^^
  File "/Users/sandeepmalkanoor/Documents/Python/airflow_env/lib/python3.11/site-packages/pandas/io/sql.py", line 970, in exists
    return self.pd_sql.has_table(self.name, self.schema)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sandeepmalkanoor/Documents/Python/airflow_env/lib/python3.11/site-packages/pandas/io/sql.py", line 2865, in has_table
    return len(self.execute(query, [name]).fetchall()) > 0
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sandeepmalkanoor/Documents/Python/airflow_env/lib/python3.11/site-packages/pandas/io/sql.py", line 2672, in execute
    cur = self.con.cursor()
          ^^^^^^^^^^^^^^^
AttributeError: 'Engine' object has no attribute 'cursor'

2025-05-13 04:39:01,516 | ERROR | ‚ùå A/R Aging extraction failed: 'Engine' object has no attribute 'cursor'
Traceback (most recent call last):
  File "/Users/sandeepmalkanoor/Documents/Python/Night_Audit_DataEngineering_Project/night_audit_etl_pipeline/processor.py", line 92, in handle_section
    insert_dataframe(engine, df, table_name, filename)
  File "/Users/sandeepmalkanoor/Documents/Python/Night_Audit_DataEngineering_Project/night_audit_etl_pipeline/db_utils.py", line 56, in insert_dataframe
    df.to_sql(table_name, con=engine, if_exists='append', index=False)
  File "/Users/sandeepmalkanoor/Documents/Python/airflow_env/lib/python3.11/site-packages/pandas/util/_decorators.py", line 333, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sandeepmalkanoor/Documents/Python/airflow_env/lib/python3.11/site-packages/pandas/core/generic.py", line 3087, in to_sql
    return sql.to_sql(
           ^^^^^^^^^^^
  File "/Users/sandeepmalkanoor/Documents/Python/airflow_env/lib/python3.11/site-packages/pandas/io/sql.py", line 842, in to_sql
    return pandas_sql.to_sql(
           ^^^^^^^^^^^^^^^^^^
  File "/Users/sandeepmalkanoor/Documents/Python/airflow_env/lib/python3.11/site-packages/pandas/io/sql.py", line 2850, in to_sql
    table.create()
  File "/Users/sandeepmalkanoor/Documents/Python/airflow_env/lib/python3.11/site-packages/pandas/io/sql.py", line 984, in create
    if self.exists():
       ^^^^^^^^^^^^^
  File "/Users/sandeepmalkanoor/Documents/Python/airflow_env/lib/python3.11/site-packages/pandas/io/sql.py", line 970, in exists
    return self.pd_sql.has_table(self.name, self.schema)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sandeepmalkanoor/Documents/Python/airflow_env/lib/python3.11/site-packages/pandas/io/sql.py", line 2865, in has_table
    return len(self.execute(query, [name]).fetchall()) > 0
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sandeepmalkanoor/Documents/Python/airflow_env/lib/python3.11/site-packages/pandas/io/sql.py", line 2672, in execute
    cur = self.con.cursor()
          ^^^^^^^^^^^^^^^
AttributeError: 'Engine' object has no attribute 'cursor'

2025-05-13 04:39:01,696 | ERROR | ‚ùå A/R Aging extraction failed: 'Engine' object has no attribute 'cursor'
Traceback (most recent call last):
  File "/Users/sandeepmalkanoor/Documents/Python/Night_Audit_DataEngineering_Project/night_audit_etl_pipeline/processor.py", line 92, in handle_section
    insert_dataframe(engine, df, table_name, filename)
  File "/Users/sandeepmalkanoor/Documents/Python/Night_Audit_DataEngineering_Project/night_audit_etl_pipeline/db_utils.py", line 56, in insert_dataframe
    df.to_sql(table_name, con=engine, if_exists='append', index=False)
  File "/Users/sandeepmalkanoor/Documents/Python/airflow_env/lib/python3.11/site-packages/pandas/util/_decorators.py", line 333, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sandeepmalkanoor/Documents/Python/airflow_env/lib/python3.11/site-packages/pandas/core/generic.py", line 3087, in to_sql
    return sql.to_sql(
           ^^^^^^^^^^^
  File "/Users/sandeepmalkanoor/Documents/Python/airflow_env/lib/python3.11/site-packages/pandas/io/sql.py", line 842, in to_sql
    return pandas_sql.to_sql(
           ^^^^^^^^^^^^^^^^^^
  File "/Users/sandeepmalkanoor/Documents/Python/airflow_env/lib/python3.11/site-packages/pandas/io/sql.py", line 2850, in to_sql
    table.create()
  File "/Users/sandeepmalkanoor/Documents/Python/airflow_env/lib/python3.11/site-packages/pandas/io/sql.py", line 984, in create
    if self.exists():
       ^^^^^^^^^^^^^
  File "/Users/sandeepmalkanoor/Documents/Python/airflow_env/lib/python3.11/site-packages/pandas/io/sql.py", line 970, in exists
    return self.pd_sql.has_table(self.name, self.schema)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sandeepmalkanoor/Documents/Python/airflow_env/lib/python3.11/site-packages/pandas/io/sql.py", line 2865, in has_table
    return len(self.execute(query, [name]).fetchall()) > 0
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sandeepmalkanoor/Documents/Python/airflow_env/lib/python3.11/site-packages/pandas/io/sql.py", line 2672, in execute
    cur = self.con.cursor()
          ^^^^^^^^^^^^^^^
AttributeError: 'Engine' object has no attribute 'cursor'

2025-05-13 04:39:01,769 | ERROR | ‚ùå A/R Aging extraction failed: 'Engine' object has no attribute 'cursor'
Traceback (most recent call last):
  File "/Users/sandeepmalkanoor/Documents/Python/Night_Audit_DataEngineering_Project/night_audit_etl_pipeline/processor.py", line 92, in handle_section
    insert_dataframe(engine, df, table_name, filename)
  File "/Users/sandeepmalkanoor/Documents/Python/Night_Audit_DataEngineering_Project/night_audit_etl_pipeline/db_utils.py", line 56, in insert_dataframe
    df.to_sql(table_name, con=engine, if_exists='append', index=False)
  File "/Users/sandeepmalkanoor/Documents/Python/airflow_env/lib/python3.11/site-packages/pandas/util/_decorators.py", line 333, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sandeepmalkanoor/Documents/Python/airflow_env/lib/python3.11/site-packages/pandas/core/generic.py", line 3087, in to_sql
    return sql.to_sql(
           ^^^^^^^^^^^
  File "/Users/sandeepmalkanoor/Documents/Python/airflow_env/lib/python3.11/site-packages/pandas/io/sql.py", line 842, in to_sql
    return pandas_sql.to_sql(
           ^^^^^^^^^^^^^^^^^^
  File "/Users/sandeepmalkanoor/Documents/Python/airflow_env/lib/python3.11/site-packages/pandas/io/sql.py", line 2850, in to_sql
    table.create()
  File "/Users/sandeepmalkanoor/Documents/Python/airflow_env/lib/python3.11/site-packages/pandas/io/sql.py", line 984, in create
    if self.exists():
       ^^^^^^^^^^^^^
  File "/Users/sandeepmalkanoor/Documents/Python/airflow_env/lib/python3.11/site-packages/pandas/io/sql.py", line 970, in exists
    return self.pd_sql.has_table(self.name, self.schema)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sandeepmalkanoor/Documents/Python/airflow_env/lib/python3.11/site-packages/pandas/io/sql.py", line 2865, in has_table
    return len(self.execute(query, [name]).fetchall()) > 0
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sandeepmalkanoor/Documents/Python/airflow_env/lib/python3.11/site-packages/pandas/io/sql.py", line 2672, in execute
    cur = self.con.cursor()
          ^^^^^^^^^^^^^^^
AttributeError: 'Engine' object has no attribute 'cursor'

2025-05-13 04:39:01,960 | ERROR | ‚ùå A/R Aging extraction failed: 'Engine' object has no attribute 'cursor'
Traceback (most recent call last):
  File "/Users/sandeepmalkanoor/Documents/Python/Night_Audit_DataEngineering_Project/night_audit_etl_pipeline/processor.py", line 92, in handle_section
    insert_dataframe(engine, df, table_name, filename)
  File "/Users/sandeepmalkanoor/Documents/Python/Night_Audit_DataEngineering_Project/night_audit_etl_pipeline/db_utils.py", line 56, in insert_dataframe
    df.to_sql(table_name, con=engine, if_exists='append', index=False)
  File "/Users/sandeepmalkanoor/Documents/Python/airflow_env/lib/python3.11/site-packages/pandas/util/_decorators.py", line 333, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sandeepmalkanoor/Documents/Python/airflow_env/lib/python3.11/site-packages/pandas/core/generic.py", line 3087, in to_sql
    return sql.to_sql(
           ^^^^^^^^^^^
  File "/Users/sandeepmalkanoor/Documents/Python/airflow_env/lib/python3.11/site-packages/pandas/io/sql.py", line 842, in to_sql
    return pandas_sql.to_sql(
           ^^^^^^^^^^^^^^^^^^
  File "/Users/sandeepmalkanoor/Documents/Python/airflow_env/lib/python3.11/site-packages/pandas/io/sql.py", line 2850, in to_sql
    table.create()
  File "/Users/sandeepmalkanoor/Documents/Python/airflow_env/lib/python3.11/site-packages/pandas/io/sql.py", line 984, in create
    if self.exists():
       ^^^^^^^^^^^^^
  File "/Users/sandeepmalkanoor/Documents/Python/airflow_env/lib/python3.11/site-packages/pandas/io/sql.py", line 970, in exists
    return self.pd_sql.has_table(self.name, self.schema)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sandeepmalkanoor/Documents/Python/airflow_env/lib/python3.11/site-packages/pandas/io/sql.py", line 2865, in has_table
    return len(self.execute(query, [name]).fetchall()) > 0
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sandeepmalkanoor/Documents/Python/airflow_env/lib/python3.11/site-packages/pandas/io/sql.py", line 2672, in execute
    cur = self.con.cursor()
          ^^^^^^^^^^^^^^^
AttributeError: 'Engine' object has no attribute 'cursor'

